app/controllers/generic_object_controller.rb

```diff
@@ -2,6 +2,8 @@ class GenericObjectController < ApplicationController
   before_action :check_privileges
   before_action :get_session_data
 
+  before_action :create_display_methods, :only => [:show]
```

Comment:
Please, do not use `before_action`. Just call the method where it is needed. I believe we already discussed this. It really makes the code far less readable than it could be.

Reply:
I think it's just a personal preference to declare the method this way. 
I somehow prefer to keep it concealed, the intention being - not to add too many things in `show` 

But if you prefer an explicit `show`, I can change that.

Reply:
@AparnaKarve `before_action` adds extra complexity and non-linearity to the code that is hard to follow. Especially if you're using it for just a single controller method, it doesn't even make sense, just complicates the life of anyone who wants to understand it later.